pkg/engine/workflow/engine.go

```diff
@@ -166,13 +174,14 @@ func Execute(pl *ActivePlan, em *engine.Metadata, c client.Client, enh renderer.
 				// stopped in the spirit of "fail-loud-and-proud".
 				switch {
 				case errors.Is(err, engine.ErrFatalExecution):
-					phaseStatus.Status = v1beta1.ExecutionFatalError
-					stepStatus.Status = v1beta1.ExecutionFatalError
-					planStatus.Status = v1beta1.ExecutionFatalError
+					phaseStatus.SetStatusWithMessage(v1beta1.ExecutionFatalError, err.Error())
```

Comment:
thinking out loud: it was ok to propagate an `ExecutionFatalError` up, but do we want to do this for the message too? It seems somewhat verbose and redundant in the output. Wdyt?

Reply:
Yeah I was not sure here as well... probably keeping it at the lowest level and not propagating up could be a good enough solution? Let's hear from @mpereira since he requested this

Reply:
Would a plan's error message be an aggregation of its phases/steps error messages in case of multiple errored steps? Maybe it makes sense keeping error messages in the leaf steps? Assuming a possible future DAG plan engine, that would probably be the way to go, right?

Reply:
yep ... so if I understand it correctly we're all leaning toward propagating status but not propagating message. I'll adjust the PR...